### DISCUSSION TITLE
Input for Source via Sink

### DISCUSSION THREAD

**Discussion by average-gary** [vectordotdev/vector#21854] [Q&A] [answered]
Let's say I have 30k devices that I'd like to hit with prometheus. The list of IPs is queryable via an HTTP endpoint.

How can I hit the HTTP endpoint and have that list of IPs be used for the prometheus list of endpoint?

**Comment by jszwedko** [comment, marked as answer]
Hi @average-gary ,

Unfortunately I don't think there is any way to do this natively in Vector. I think the best approach would be to have an external process fetch the IPs and write out a Vector configuration. If the list is dynamic, the external process could run on an interval and send a `SIGHUP` to Vector to have it reload its configuration.

**Reply by average-gary** [reply]
ack. Thank you.
